src/test/java/com/microsoft/sqlserver/jdbc/datatypes/DateAndTimeTypeTest.java

```diff
@@ -0,0 +1,204 @@
+/*
+ * Microsoft JDBC Driver for SQL Server
+ * 
+ * Copyright(c) Microsoft Corporation All rights reserved.
+ * 
+ * This program is made available under the terms of the MIT License. See the LICENSE file in the project root for more information.
+ */
+package com.microsoft.sqlserver.jdbc.datatypes;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assumptions.assumeTrue;
+
+import java.sql.Connection;
+import java.sql.Date;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.sql.Time;
+import java.sql.Timestamp;
+
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.platform.runner.JUnitPlatform;
+import org.junit.runner.RunWith;
+import org.opentest4j.TestAbortedException;
+
+import com.microsoft.sqlserver.jdbc.SQLServerDataTable;
+import com.microsoft.sqlserver.jdbc.SQLServerPreparedStatement;
+import com.microsoft.sqlserver.jdbc.SQLServerStatement;
+import com.microsoft.sqlserver.testframework.AbstractTest;
+import com.microsoft.sqlserver.testframework.DBConnection;
+import com.microsoft.sqlserver.testframework.Utils;
+
+@RunWith(JUnitPlatform.class)
+public class DateAndTimeTypeTest extends AbstractTest {
+
+    private static final Date DATE_TO_TEST = new java.sql.Date(2017, 20, 11);
```

Comment:
Could we use the non-deprecated constructors for Date, Time and Timestamp?

Reply:
I think this style of argument is deprecated, and they want us to provide a long value that corresponds to this time instead. The long values for these Date, Time and Timestamp objects come to 61494793200000L, 74096000L, and 61494838496000L respectively, I believe.